.vscode/launch.json

```diff
@@ -0,0 +1,39 @@
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Python: Attach",
+      "type": "python",
+      "request": "attach",
+      "connect": {
+        "host": "localhost",
+        "port": 6789
+      },
+      "pathMappings": [
+        {
+          "localRoot": "${workspaceFolder}/backend",
+          "remoteRoot": "/app/backend"
+        }
+      ],
+
+    },
+    {
+      "name": "Docker: flask-dev - save_routes.py",
+      "type": "docker",
+      "request": "launch",
+      "preLaunchTask": "docker-run: save_routes.py",
+      "python": {
+        "pathMappings": [
+          {
+            "localRoot": "${workspaceFolder}/backend",
+            "remoteRoot": "/app/backend"
+          }
+        ],
+        "projectType": "general"
+      }
+    }
```

Comment:
Could you explain the purpose of this Docker launch config? I would have assumed the `Python: Attach` is all we would have needed.

Reply:
Definitely! The `Python: Attach` launch config I added is, at this point, just used in conjunction with the docker-compose file added in this PR to be able to debug the flask app. This block plus the code in `tasks.json` is used to build and run a `flask-dev` container that calls the `save_routes.py` script. (In one of my comments above, I mentioned we might want to expand this to accept user input for the specific cli script to run, but I started with `save_routes`, since that's what I was working on in #42.) This config makes it possible to launch a container that will call that script in one click from the debug and run sidebar.

I think it would be possible to use the `Python: Attach` config if we wanted to have devs start a shell via `./docker-shell.sh`, then `pip install debugpy`, then add something like the block in this [SO post](https://stackoverflow.com/questions/64293300/vscode-debugger-with-docker-compose#answer-64378331) to their script, and then call the script. And we'd want to open the debug port in the `Dockerfile` as well.

In writing this up, and thinking about your comment below, it feels like there should be some solution where we'd have an entrypoint script for starting up an image that takes care of installing `debugpy` and calling it with the desired port to listen on, and then being able to pass in the command to run. Basically, it would be great if we could `docker compose up` with some reference to an optional entrypoint script and also optionally pass in the desired command (either `flask run`, or a specified script). I have not played around with that though, so I'm not sure how straightforward it is to build in that level of flexibility, without overcomplicating the process for someone who just wants to `docker compose up` the dev stack.

Reply:
Ok! Just getting back to this after having been out of town (and preparing to leave town again 😬 ). So I'm not sure how much the addition of `debugpy` to the `flask-dev` image buys us, in terms of file / code reduction. We'd still need some way to override the default command when starting up that container (and we'd need port `6789` in the compose service definition - but that's a fairly inconsequential addition).

Having messed around with this a little more, I think we could at least reduce the contents of the `docker-compose.debug.yml` file, if we wanted to go the route of having folks run `docker compose up` with multiple compose files (kind of like the example I added to the docs for including the `docker-compose.overrides.yml` file, but the debug file would be added too). The debug compose file would just extend the definition from the base `docker-compose.yml` file. How does that sound?

Reply:
Alright, after trying this out a little more, I think it would work to have the following in the `docker-compose.debug.yml` file:
```
version: "3.7"
services:
  flask-dev:
    command: ["sh", "-c", "pip install debugpy -t /tmp && python /tmp/debugpy --wait-for-client --listen 0.0.0.0:6789 -m flask run --no-debugger --no-reload --host 0.0.0.0 --port 5000"]
    ports:
      - 6789:6789
```
And then the debug compose config could be started up with something like `docker compose -f "docker-compose.yml" -f "docker-compose.debug.yml" -f "docker-compose.override.yml" up --build`

That would at least reduce the amount of duplicative code between the base and the debug compose files, and keep the vs code debug option opt-in rather than opt-out. I'm definitely open to other thoughts or suggestions though!